File: SdkTestAutomation.CliWrappers/SdkTestAutomation.Python/sdk_wrapper/operation_utils.py
#!/usr/bin/env python3

import os
from typing import Callable, Any
from conductor.client.http.api_client import ApiClient
from .sdk_response import SdkResponse


def execute_with_error_handling(operation: Callable[[], SdkResponse]) -> SdkResponse:
    """Execute an operation with centralized error handling"""
    try:
        return operation()
    except Exception as e:
        return SdkResponse.create_error(500, str(e))


def create_sdk_configuration() -> ApiClient:
    """Create SDK configuration with server URL from environment variable or default"""
    api_client = ApiClient()
    server_url = os.getenv('CONDUCTOR_SERVER_URL')
    if server_url:
        api_client.configuration.host = server_url
    return api_client 

File: SdkTestAutomation.CliWrappers/SdkTestAutomation.Java/src/main/java/com/conductor/sdkwrapper/OperationUtils.java
package com.conductor.sdkwrapper;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.netflix.conductor.client.http.ConductorClient;

public class OperationUtils {
    
    public static SdkResponse executeWithErrorHandling(OperationSupplier operation) {
        try {
            return operation.execute();
        } catch (Exception e) {
            return SdkResponse.createError(500, e.getMessage());
        }
    }
    
    public static ConductorClient createSdkConfiguration() {
        String serverUrl = System.getenv("CONDUCTOR_SERVER_URL");
        if (serverUrl == null) {
            serverUrl = "http://localhost:8080/api";
        }
        return new ConductorClient.Builder().basePath(serverUrl).build();
    }
    
    @FunctionalInterface
    public interface OperationSupplier {
        SdkResponse execute() throws Exception;
    }
} 